SCHEMA.md

```diff
@@ -0,0 +1,84 @@
+
+In research, some responses are typically discarded (e.g., outliers). The protocol does not enforce consideration of all responses. Confirmation of receipt is possible, but confirmation of consideration is not required. The researcher is trusted to tally the results reasonably. They may optionally publish aggregated survey results in the same Waku topic but are not required to prove how responses were handled.
+
+### Roles in the protocol:
+- **Researcher**: Creates and distributes surveys, collects responses, and optionally publishes results.
+- **Respondent**: Receives survey notifications, submits encrypted responses.
+
+### Protocol Flow:
+1. The Researcher posts a `SurveyCreate` message over Waku.
+2. The Respondent:
+   - Decodes the `SurveyCreate` message.
+   - Generates a `SurveyResponse`, encrypting responses with the survey's public key.
+   - Posts the `SurveyResponse` to the Waku topic.
+3. The Researcher collects and aggregates responses.
+4. *(Optional)* The Researcher posts a `SurveyClose` message to close the survey and indicate a deadline.
+5. *(Optional)* The Researcher posts a `SurveyResults` message with aggregated results.
+
+### Encryption Modes
+There are three security modes:
+1. **Unencrypted**: All data is transparent. Generally not recommended but may work for non-sensitive use cases. Respondents should avoid including personally identifiable data.
+2. **Encrypted responses**: The survey is public, but responses are encrypted using the researcher's public key. Only the researcher can decrypt responses.
+3. **Encrypted and authenticated responses**: In addition to encryption, respondents include a proof of eligibility in their responses. This allows the researcher to limit participants, e.g., requiring a wallet in the response. Advanced setups may use zero-knowledge proofs of membership or signatures proving ownership of a token (NFT, Ordinal).
+
+---
+
+```proto
+syntax = "proto3";
+
+message SurveyCreate {
+    string survey_id = 1;
+    repeated FormQuestion survey_questions = 2;
+    bytes survey_pub_key = 3;
+    optional string survey_eligibility_requirements = 4;
+    optional uint64 survey_deadline = 5;
+}
+
+message FormQuestion {
+    enum SurveyQuestionType {
+        TEXT = 0;
+        SINGLE_CHOICE = 1;
+        MULTI_CHOICE = 2;
+    }
+    SurveyQuestionType question_type = 1;
+    string survey_question = 2;
+    repeated string survey_options = 3; // Optional
+}
+
+message SurveyResponse {
+    string response_id = 1;
+    bytes encrypted_survey_response_payload = 2;
+}
+
+message SurveyResponsePayload {
+    string survey_id = 1;
+    bytes survey_response = 2;
```

Comment:
Do we need to specify this similarly to `FormQuestion`? 

As in there is a `FormQuestionResponse` with question index as an ID and answer (string or choice index)?

Reply:
Great idea, here is a new version [04635cb](https://github.com/whisperbox-org/whisperbox/pull/13/commits/04635cbb24e63709f984a3de63905a1ee669c67c)